buffer-api/src/main/java/io/netty/buffer/api/adaptor/ByteBufAdaptor.java

```diff
@@ -1416,11 +1419,8 @@ public ByteBuf slice(int index, int length) {
     @Override
     public ByteBuf retainedSlice(int index, int length) {
         checkAccess();
-        try {
-            return wrap(buffer.slice(index, length));
-        } catch (IllegalStateException e) {
-            throw new IllegalReferenceCountException(e);
-        }
+        retain();
```

Comment:
is there any chance that ` new Slice` might throw ? If so would it be better to call retain() after it ?